### ISSUE TITLE
joist buttons do not use enabledProperty

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/joist#676]
In https://github.com/phetsims/sun/issues/515#issuecomment-720709179 (sun buttons have a different API for enabledProperty), @arouinfar reported:

> I reviewed this in the context of Gravity and Orbits, and compared master to the most recently QA-tested version, [1.2.0-dev.16](https://phet-dev.colorado.edu/html/gravity-and-orbits/1.2.0-dev.16/phet-io/wrappers/studio/). Generally, things are looking really nice and the tree seems undisturbed. The only difference I noticed was that setting the `enabledProperty` of the buttons in the navbar to false doesn't remove the hover behavior. The vertical lines appear on either side of the button even though it is not pickable. This did not occur previously, and I think the hover behavior creates the illusion that the buttons are still enabled. Affected phetioIDs:
* `gravityAndOrbits.general.view.navigationBar.homeButton.enabledProperty`
* `gravityAndOrbits.general.view.navigationBar.modelScreenButton.enabledProperty`
* `gravityAndOrbits.general.view.navigationBar.toScaleScreenButton.enabledProperty`
* `gravityAndOrbits.general.view.navigationBar.phetButton.enabledProperty`

**Comment by pixelzoom** [comment]
Re navigation bar buttons...  

* The buttons in the navigation bar are not sun buttons.  The home screen button (HomeButton) is a subclass of JoistButton, which is an entirely different button implementation.  The screen buttons are subclasses of Node.

* In Natural Selection https://phet-dev.colorado.edu/html/natural-selection/1.2.0-rc.2/phet-io/wrappers/studio/, published well before work started on sun buttons, I see the same behavior for the Intro and Lab navigation bar buttons.  If they are disabled, the highlight still appears on pointer over.   That does not occur for the home button, which is different than what @arouinfar described above for GAO.

* None of these buttons currently do anything specific with enabledProperty.  At a minimum, it sounds like they need to disable their highlight.

Assigning to @ariel-phet to prioritize and assign.

**Comment by pixelzoom** [comment]
I'm not clear (in general) with what happens with focus for disabled components. What should happen with focusHighlight when a navigation bar button is disabled?

**Comment by ariel-phet** [comment]
@zepumph this seems to be your wheelhouse as it has overlap with iO/a11y/recent work on enabled property.

Marking as high priority since it seems like this work is in the "wrapping up" phase and it would be good to iron out the details. Let me know if that priority should be reassessed.

**Comment by zepumph** [comment]
>I'm not clear (in general) with what happens with focus for disabled components. What should happen with focusHighlight when a navigation bar button is disabled? And if the button had focus when it because disabled, where does focus go?

In general disabled elements are still focusable. It is an alt-input no-no to remove or change the focus order, as it is very disorienting. You can focus, but you can't activate a button.

**Comment by zepumph** [comment]
This was solved above. I don't feel like I did anything but a bug fix. The long-term, (joist 2.0), solution would be to consolidate these joist button types, and make them extend from sun buttons, and behave similarly. For now though, I just tapped into the buttonModel's enabledProperty to achieve this regression fix. @pixelzoom  please review.

I'm not too interested in adding more button refactoring to my plate right now. But I'll assign @ariel-phet if he wants more done with making Joist buttons better as part of this issue or a more general 2.0 issue.

**Comment by ariel-phet** [comment]
I think that is good enough @zepumph but tagging this issue as part of https://github.com/phetsims/joist/issues/678

**Comment by zepumph** [comment]
Sounds good. Over to @pixelzoom for review.

**Comment by pixelzoom** [comment]
Code changes look reasonable.  Over to @arouinfar to verify that this is now behaving as desired, since she reported this problem.

**Comment by arouinfar** [comment]
Looks good in master. Thanks @pixelzoom @zepumph.
